src/plugins/point_in_time_management/public/components/pit_table/pit_table.tsx

```diff
@@ -0,0 +1,182 @@
+    fetchDataSources();
+  });
+
+  const fetchDataSources = () => {
+    getDataSources(savedObjects.client)
+      .then((fetchedDataSources) => {
+        if (fetchedDataSources?.length) {
+          setDataSources(
+            fetchedDataSources
+              .concat([defaultDataSource])
+              .sort((a, b) => a.sort.localeCompare(b.sort))
+          );
+        }
+      })
+      .catch(() => {
+        toasts.addDanger(
+          i18n.translate('pitManagement.pitTable.fetchDataSourceError', {
+            defaultMessage: 'Unable to find existing data sources',
+          })
+        );
+      });
+  };
+
+  const columns = [
+    {
+      field: 'name',
+      name: i18n.translate('pitManagement.pitTable.nameColumnName', {
+        defaultMessage: 'Name',
+      }),
+    },
+    {
+      field: 'expires',
+      name: i18n.translate('pitManagement.pitTable.expiresColumnName', {
+        defaultMessage: 'Expires',
+      }),
+      sortable: true,
+    },
+    {
+      field: 'dataSource',
+      name: i18n.translate('pitManagement.pitTable.dataSourceColumnName', {
+        defaultMessage: 'Data Source',
+      }),
+    },
+    {
+      field: 'created',
+      name: i18n.translate('pitManagement.pitTable.createdColumnName', {
+        defaultMessage: 'Created',
+      }),
+      sortable: true,
+    },
+    {
+      field: 'actions',
+      name: i18n.translate('pitManagement.pitTable.actionsColumnName', {
+        defaultMessage: 'Actions',
+      }),
+    },
+  ];
+
+  const renderToolsRight = () => {
+    return [
+      <EuiButton
+        iconType="trash"
+        key="loadUsers"
+        isDisabled={pits.length === 0}
+        data-test-subj="deletePITBtnInPitTable"
+      >
+        <FormattedMessage id="pitManagement.pitTable.deletePitButton" defaultMessage="Delete" />
+      </EuiButton>,
+    ];
+  };
+
+  const search: EuiSearchBarProps = {
+    toolsRight: renderToolsRight(),
+    defaultQuery: 'dataSource:Default',
+
+    box: {
+      incremental: true,
+      schema: true,
+      disabled: pits.length === 0,
+    },
+    filters: [
+      {
+        type: 'field_value_selection',
+        searchThreshold: 5,
+        field: 'dataSource',
+        name: i18n.translate('pitManagement.pitTable.dataSourceFilterName', {
+          defaultMessage: 'Data Source',
+        }),
+        multiSelect: false,
+        options: dataSources.map((source) => ({
+          value: source.title,
+          name: source.title,
+          view: `${source.title}`,
+        })),
+      },
+    ],
+  };
+
+  const pagination = {
+    initialPageSize: 10,
+    pageSizeOptions: [5, 10, 25, 50],
+  };
+
+  return (
+    <>
+      <EuiPageContent
+        className="pitTable"
+        horizontalPosition="center"
+        data-test-subj="pointInTimeTable"
+      >
+        <PageHeader />
+        <EuiSpacer size="m" />
+        <EuiPageContentBody>
+          <EuiInMemoryTable
+            items={pits}
+            itemId="name"

```

Comment:
I see itemId="id" used in other pages - can we use the same for consistency?

Reply:
Will update this in following PR based on the property name we have in the PIT object.